src/components/Reports/SalaryReport/SalaryReportTable.tsx

```diff
@@ -1,10 +1,238 @@
+      <TableCell colSpan={16}>
+        <Box
+          height="100%"
+          alignItems="center"
+          justifyContent="center"
+          bgcolor={colors.green[600]}
+        >
+          Loading
+        </Box>
+      </TableCell>
+    </TableRow>
+  );
+
+  const renderEmpty = () => (
+    <TableRow>
+      <TableCell colSpan={16}>No Data</TableCell>
+    </TableRow>
+  );
+
+  const renderError = () => (
+    <TableRow>
+      <TableCell colSpan={16}>
+        <Box bgcolor={colors.red[600]}>Error: {error?.toString()}</Box>
+      </TableCell>
+    </TableRow>
+  );
+
+  return (
+    <Box>
+      <Box my={2}>
+        <Grid
+          container
+          justify="space-between"
+          className={clsx(classes.root, className)}
+          {...rest}
+        >
+          <Grid item>
+            <Typography variant="h5">{t(title)}</Typography>
+          </Grid>
+          <Grid item>
+            <ButtonGroup aria-label="report header button group">
+              <Button
+                startIcon={
+                  <SvgIcon fontSize="small">
+                    <CodeIcon />
+                  </SvgIcon>
+                }
+              >
+                {t('Expand Partner Info')}
+              </Button>
+              <Button
+                startIcon={
+                  <SvgIcon fontSize="small">
+                    <GetAppIcon />
+                  </SvgIcon>
+                }
+              >
+                <CSVLink
+                  data={contacts}
+                  filename={`mpdx-salary-contributions-export-${DateTime.now().toISODate()}.csv`}
+                  className={classes.downloadCsv}
+                >
+                  {t('Export')}
+                </CSVLink>
+              </Button>
+              <Button
+                startIcon={
+                  <SvgIcon fontSize="small">
+                    <PrintIcon />
+                  </SvgIcon>
+                }
+              >
+                {t('Print')}
+              </Button>
+            </ButtonGroup>
+          </Grid>
+        </Grid>
+      </Box>
+      <TableContainer>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>
+                <Typography variant="h6">{salaryCurrency}</Typography>
+              </TableCell>
+              {currencyGroups?.map((currencyGroup) => {
+                const allYears = currencyGroup.totals.months.map(
+                  (month) => month.month.split('-')[0],
+                );
+                const monthCount = allYears.reduce(
+                  (count, year) => ({
+                    ...count,
+                    [year]:
+                      ((count as { [key: string]: number })[year] || 0) + 1,
+                  }),
+                  {},
+                );
+
+                return Object.entries(monthCount).map(([year, count]) => (
+                  <TableCell
+                    key={year}
+                    colSpan={count as number}
```

Comment:
Should be typed now that the reduce is typed.
```suggestion
                    colSpan={count}
```